app/src/main/java/com/github/polypoly/app/ui/menu/rankings/RankingsViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.github.polypoly.app.ui.menu.rankings
+
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.lifecycle.ViewModel
+import com.github.polypoly.app.base.user.User
+import com.github.polypoly.app.network.getAllValues
+import com.github.polypoly.app.utils.global.GlobalInstances.Companion.remoteDB
+import com.github.polypoly.app.utils.global.Settings.Companion.DB_USERS_PROFILES_PATH
+
+/**
+ * The view model of the rankings activity
+ *
+ * @see RankingsActivity
+ */
+class RankingsViewModel: ViewModel() {
+    /**
+     * The list of users to rank
+     */
+    private var users: List<User> = emptyList()
+
+    /**
+     * The rankings of the users according to the different categories
+     *
+     * @see RankingCategory
+     */
+    private val rankings: Map<RankingCategory, SnapshotStateList<User>> = enumValues<RankingCategory>()
+        .associateWith { SnapshotStateList() }
+
+    /**
+     * Fetches the users from the database and recomputes the rankings
+     */
+    fun fetchUsers() {
+        // Request all registered users from the database
+        remoteDB.getAllValues<User>(DB_USERS_PROFILES_PATH).thenAccept {
```

Comment:
As on a further PR we'll encapsulate all DB operations, please write a TODO on all lines where the DB is explicitly called

Reply:
Noted!